File: backend/src/main/java/ecrow/backend/business/abstracts/ContactMessageService.java
package ecrow.backend.business.abstracts;

import ecrow.backend.core.utilities.results.DataResult;
import ecrow.backend.core.utilities.results.Result;
import ecrow.backend.entities.concretes.ContactMessage;
import ecrow.backend.entities.dtos.ContactMessageAddDto;

import java.util.List;

public interface ContactMessageService {
    DataResult<List<ContactMessage>> getAll();
    DataResult<ContactMessage> getById(Integer id);
    Result deleteById(Integer id);
    Result deleteAll();

    Result add(ContactMessageAddDto contactMessageAddDto);
}

File: frontend/src/services/ContactMessageService.js
import axios from "axios";

export default class ContactMessageService{

    getAll(){
        return axios.get("http://localhost:8080/api/contactMessage/getAll")
    }

    getById(id){
        return axios.get(`http://localhost:8080/api/contactMessage/getById?id=${id}`)
    }

    deleteAll(){
        return axios.delete("http://localhost:8080/api/contactMessage/deleteAll")
    }

    deleteById(id){
        return axios.delete(`http://localhost:8080/api/contactMessage/deleteById?id=${id}`)
    }

    add(values){
        return axios.post("http://localhost:8080/api/contactMessage/add", values)
    }
}